packages/ember-template-compiler/lib/plugins/transform-big-number-into-parse-int-helper.ts

```diff
@@ -0,0 +1,51 @@
+import { AST, ASTPlugin, ASTPluginEnvironment } from '@glimmer/syntax';
+
+/**
+ @module ember
+*/
+
+/**
+  A Glimmer2 AST transformation that replaces all instances of
+
+  ```handlebars
+ {{foo 12345566788}}
+  ```
+
+  to
+
+  ```handlebars
+ {{foo (-parse-int "12345566788")}}
+  ```
+
+  @private
+  @class TransformBigNumberIntoParseIntHelper
+*/
+
+export default function transformBigNumberIntoParseIntHelper(env: ASTPluginEnvironment): ASTPlugin {
+  let { builders: b } = env.syntax;
+
+  return {
+    name: 'transform-big-number-into-parse-int-helper',
+
+    visitor: {
+
+      NumberLiteral(number: AST.NumberLiteral): AST.Node {
+        //See https://github.com/emberjs/ember.js/issues/15635 for the choice for [9].
+        if(String(number.original).length > 9) {
```

Comment:
Needs to use `length > 8` The bug can occur with values >= 536870912 and < 1000000000, for example 678881280.

Reply:
I think we should just parse the number with `parseInt` and test the actual value, instead of the length of the string

Reply:
We need to check for floats as well, maybe just use `parseFloat` here, check that it is an integer (`parseFloat(...) % 1 === 0`?) and is out of bounds, both the upper and lower (negative) bounds.

Reply:
or `new Number('1').valueOf()`

Reply:
> Overall looks good to me, left some comments for things that need to be fixed before merging

@chancancode @lifeart @Mooninaut, 

Thanks for the comments...
 
Given that it was an old bug, I should have checked the replication before attempting the PR; my bad. 

Here are the results though:

From the [twiddle](https://ember-twiddle.com/98647deda69ba3c58a58b1c6decb0add?openFiles=templates.application.hbs%2C) shared by @rwjblue in the bug description, I had 

```mustache
{{go-boom-for-big-numbers 1234567893}}
{{go-boom-for-big-numbers 12345678933}}
<div/>
{{go-boom-for-big-numbers 1234567893232323.2323}}
{{go-boom-for-big-numbers 1234567893323232323.2323}}
<div/>
{{go-boom-for-big-numbers -1234567893232323.2323}}
{{go-boom-for-big-numbers -1234567893323232323.2323}}
<div/>
{{go-boom-for-big-numbers -1234567893}}
{{go-boom-for-big-numbers -12345678933}}
```

The following are the results: 

**Ember 3.0.0**
(No display, console error) : `Uncaught TypeError: reference$$1.value is not a function`

**Ember 3.1.4 and newer**
```
160826069 12345678933
1234567893232323.2 1234567893323232300
-1234567893232323.2 -1234567893323232300
-160826069 -12345678933
```

There is also an error in the 3.1.4 results: `1234567893` is displayed as `160826069`. I think this should be treated as a glimmer-vm bug and tracked separately. 

Looks like this is fixed 'upstream' in newer versions of glimmer-vm. I was not able to locate the actual diff that fixed this in the glimmer-vm code. 

This makes PR obsolete, as the relevant fixes are done on the glimmer-vm code.   

I'd like a go-ahead and on continuation of this PR if its still needed..